api/src/main/java/org/hawkular/apm/api/model/trace/Node.java

```diff
@@ -231,6 +250,49 @@ public Node setDetails(Map<String, String> details) {
     }
 
     /**
+     * This method determines whether there is atleast one
+     * property with the supplied name.
+     *
+     * @param name The property name
+     * @return Whether a property of the supplied name is defined
+     */
+    public boolean hasProperty(String name) {
+        for (Property property : this.properties) {
```

Comment:
If Java 8 can be used, this method can be reduced to:
`return this.properties.stream().filter(p -> p.getName().equals(name)).findFirst().isPresent();`